Cleanup and config change for GPhL-specific file

rhfogh opened the pull request:
Added docker for Ubuntu:focal, Python3, Qt5,

rhfogh committed:
Fixes to mock-test GSK stratcal run

rhfogh committed:
Fixes to mock-test GSK stratcal run

rhfogh committed:
GPhL configuration updates, various

rhfogh committed:
GPhL configuration updates, various

rhfogh committed:
Cleanup. Also removed queue_model_objects_v1 reference

rhfogh committed:
Cleanup. Also removed queue_model_objects_v1 reference

rhfogh committed:
Changed to reflect new location of configuration files

rhfogh committed:
Changed to reflect new location of configuration files

marcus-oscarsson commented on dev/gphl/Debian9_qt4/start_mxcube2:
```diff
@@ -7,12 +7,12 @@ CONTAINER=mx3_deb9_qt4_gphl2
 # CUSTOM_BRICKS_PATH="/MXCuBE/mxcube/extra_brick_test_1:/MXCuBE/mxcube/extra_brick_test_2"
-HARDWARE_REPOSITORY_SERVER="${HO_SERVER}/gphl_mock:/${HO_SERVER}:/${HO_SERVER}/embl_hh_p14"
-CUSTOM_HARDWARE_OBJECTS_PATH=""
+HARDWARE_REPOSITORY_SERVER="${HO_SERVER}/configuration/gphl_mock:/${HO_SERVER}/test/xml-qt:${HO_SERVER}/configuration/alba_xaloc13"
```
This path is ALBA specific, is that the intention ?

rhfogh replied:
Sort of, yes. This is inside the 'gphl/' directory, which makes it global-phasing specific. Since I test relative to different beamlines  I need to have varying specific directories on the lookup path, and it seemed easier to have all the configuring in one place, and checked in, and then change it at need. That way it is visible how things work, in a way it would not be if I relied on some other mechanism, like local environment variables. But if you feel it is not a good idea to have specific locations checked in, even in a gphl/ directory, I can change to a different model

rhfogh committed:
Finished merging ALBA_gphl 2.3.0 into master.

rhfogh committed:
Finished merging ALBA_gphl 2.3.0 into master.

rhfogh committed:
Cleaned up imports, and bug fixes in latest merge

Regrettably the import cleanup (or a small part of it) was required
for the bug fix, hence the mixing of the two.

The import changes have moved to importing from HardwareRepository. etc.
and to gui. etc. in all cases, avoiding direct imports of e.g.
sample_centring, abstract.xyz, individual HOs etc.

The reason is that importing the same file under two different paths
creates two different copies in memory, so that state_x == state_y
or setting class attributes gives unpredictable results when the
class/instance is not from the same package in different cases.

rhfogh committed:
Cleaned up imports, and bug fixes in latest merge

Regrettably the import cleanup (or a small part of it) was required
for the bug fix, hence the mixing of the two.

The import changes have moved to importing from HardwareRepository. etc.
and to gui. etc. in all cases, avoiding direct imports of e.g.
sample_centring, abstract.xyz, individual HOs etc.

The reason is that importing the same file under two different paths
creates two different copies in memory, so that state_x == state_y
or setting class attributes gives unpredictable results when the
class/instance is not from the same package in different cases.

rhfogh committed:
Merge remote-tracking branch 'rhfogh/master' into rhfogh

rhfogh committed:
Fixed comment spelling

rhfogh committed:
UI and HO changes prior tio ALBA test 20190528

Added dose_budget and transmission calculation,
multiple characterisation strategies, ...

rhfogh committed:
Porting to ALBA production branch

Debugging still incomplete

rhfogh committed:
Removed api and added config for xml-qt and embl_p14. Not fully tested.

rhfogh committed:
CHanged over to ALBA config

rhfogh committed:
Fixed CODACY style complaints

rhfogh committed:
Set laptop parameters in new branch

rhfogh committed:
Set laptop parameters in new branch

marcus-oscarsson commented:
I will see you later today, if I forget, remind me to ask you about this PR :)

rhfogh commented:
@marcus-oscarsson 
As you see, this changes very slowly. 
SOme of the commits have msileading names - the reason being taht I have a tendency to do a commit to all three repositories at once, with the name reflecting the major changes (which afre in the other repositories).

The actual changes here are in two parts:
- Changes to the gphl subdirectory, which are the GPhL test dockers - free for anyone to make use of, but in practice used only by us (hence the name).
- Changes to the qt3 dockers - now long obsolete - byut unlikely to hurt anyiohne. They worked when they were made. 

Can we get this merged as it is, or what would I need to do tro get it movede forward?

rhfogh committed:
Added jsonschema, pyqtgraph

rhfogh committed:
Merge remote-tracking branch 'rhfogh/rhfogh_laptop' into rhfogh_laptop

rhfogh committed:
snake_case, some fixes, some mockup refactoring. Program now opens.

rhfogh committed:
Set config directories etc. to ALBA production values

rhfogh committed:
Change standard directory structure

rhfogh committed:
Ported GPhL code from ALBA branch, Pre-test

rhfogh committed:
Updating GPhL code

rhfogh committed:
Addressed points raised by Gleb

(that do not rquire java code changes)

rhfogh committed:
Refactored from HardwareRepository/ to mx3core/

rhfogh committed:
Revert "Refactored from HardwareRepository/ to mx3core/"

This reverts commit 2ec33193

rhfogh committed:
First cut - porting changes from P14 branch to ALBA branch

rhfogh committed:
First draft qt5 docker file

rhfogh committed:
Docker for Python3/Qt5 - WIP

rhfogh committed:
Added Ubuntu qt5 docker and updated to use it

rhfogh committed:
Merge remote-tracking branch 'origin/master' into rhfogh_laptop

rhfogh committed:
Functional Ubuntu dockers.

rhfogh committed:
Furher fixes for porting

NB some refactoring for consistency
NB Removed cp parameter for collect (for now)
